File: app/src/main/java/org/worshipsongs/activity/SongContentViewActivity.java
package org.worshipsongs.activity;


import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.WindowManager;

import org.worshipsongs.CommonConstants;
import org.worshipsongs.adapter.SongContentLandScapeViewerPageAdapter;
import org.worshipsongs.adapter.SongContentPortraitViewerPageAdapter;
import org.worshipsongs.component.SlidingTabLayout;
import org.worshipsongs.domain.Setting;
import org.worshipsongs.fragment.ISongContentPortraitViewFragment;
import org.worshipsongs.fragment.SongContentPortraitViewFragment;
import org.worshipsongs.service.PresentationScreenService;
import org.worshipsongs.service.UserPreferenceSettingService;
import org.worshipsongs.worship.R;

import java.util.ArrayList;

/**
 * @Author : Seenivasan, Madasamy, Vignesh Palanisamy
 * @Version : 1.0
 */
public class SongContentViewActivity extends AppCompatActivity
{
    private UserPreferenceSettingService userPreferenceSettingService;
    private PresentationScreenService presentationScreenService;
    private boolean isSectionView = true;
    private boolean isTabView = true;

    @Override
    public void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.song_content_view);
        userPreferenceSettingService = new UserPreferenceSettingService();
        presentationScreenService = new PresentationScreenService(this);
        if (userPreferenceSettingService.isKeepAwake()) {
            getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        }
        if (savedInstanceState != null) {
            isSectionView = savedInstanceState.getBoolean("isSectionView");
            isTabView = savedInstanceState.getBoolean("isTabView");
        }
        Intent intent = getIntent();

        ArrayList<String> titleList = intent.getExtras().getStringArrayList(CommonConstants.TITLE_LIST_KEY);
        if (Configuration.ORIENTATION_PORTRAIT == getResources().getConfiguration().orientation) {
            final SongContentPortraitViewerPageAdapter songContentPortraitViewPagerAdapter =
                    new SongContentPortraitViewerPageAdapter(getSupportFragmentManager(), titleList, presentationScreenService);
            // Assigning ViewPager View and setting the adapter
            final ViewPager pager = (ViewPager) findViewById(R.id.pager);
            pager.setAdapter(songContentPortraitViewPagerAdapter);
            // Assiging the Sliding Tab Layout View
            SlidingTabLayout tabs = (SlidingTabLayout) findViewById(R.id.tabs);
            //tabs.setVerticalScrollbarPosition();
            tabs.setDistributeEvenly(false);
            // To make the Tabs Fixed set this true, This makes the tabs Space Evenly in Available width
            // Setting Custom Color for the Scroll bar indicator of the Tab View
            tabs.setCustomTabColorizer(new SlidingTabLayout.TabColorizer()
            {
                @Override
                public int getIndicatorColor(int position)
                {
                    return getResources().getColor(android.R.color.background_dark);
                }
            });

            tabs.setVisibility(View.GONE);
            // Setting the ViewPager For the SlidingTabsLayout
            tabs.setViewPager(pager);
            pager.setCurrentItem(Setting.getInstance().getPosition());

            pager.addOnPageChangeListener(new ViewPager.OnPageChangeListener()
            {
                @Override
                public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels)
                {
                    Setting.getInstance().setPosition(position);
                }

                @Override
                public void onPageSelected(int position)
                {
                    ISongContentPortraitViewFragment fragment = (ISongContentPortraitViewFragment) songContentPortraitViewPagerAdapter.instantiateItem(pager, position);
                    if (fragment != null) {
                        fragment.fragmentBecameVisible();
                    }
                }

                @Override
                public void onPageScrollStateChanged(int state)
                {
                }
            });
        } else {
            getSupportActionBar().hide();
            SongContentLandScapeViewerPageAdapter songContentLandScapeViewerPageAdapter =
                    new SongContentLandScapeViewerPageAdapter(getSupportFragmentManager(), titleList.get(Setting.getInstance().getPosition()));
            // Assigning ViewPager View and setting the adapter
            ViewPager pager = (ViewPager) findViewById(R.id.land_pager);
            pager.setAdapter(songContentLandScapeViewerPageAdapter);
            // Assiging the Sliding Tab Layout View
            SlidingTabLayout tabs = (SlidingTabLayout) findViewById(R.id.land_tabs);
            //tabs.setVerticalScrollbarPosition();
            tabs.setDistributeEvenly(false);
            // To make the Tabs Fixed set this true, This makes the tabs Space Evenly in Available width
            // Setting Custom Color for the Scroll bar indicator of the Tab View
            tabs.setCustomTabColorizer(new SlidingTabLayout.TabColorizer()
            {
                @Override
                public int getIndicatorColor(int position)
                {
                    return getResources().getColor(android.R.color.background_dark);
                }
            });
            tabs.setVisibility(View.GONE);
            // Setting the ViewPager For the SlidingTabsLayout
            tabs.setViewPager(pager);
        }
    }

    public void onResume()
    {
        super.onResume();
        presentationScreenService.onResume();
    }


    @Override
    protected void onStop()
    {
        super.onStop();
        presentationScreenService.onStop();
    }

    @Override
    protected void onPause()
    {
        super.onPause();
        presentationScreenService.onPause();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState)
    {
        super.onSaveInstanceState(outState);
    }

}

File: app/src/main/java/org/worshipsongs/activity/SongContentViewActivity.kt
package org.worshipsongs.activity


import android.content.res.Configuration
import android.os.Bundle
import android.support.v4.view.ViewPager
import android.util.Log
import android.view.View
import android.view.WindowManager

import org.worshipsongs.CommonConstants
import org.worshipsongs.R
import org.worshipsongs.adapter.SongContentLandScapeViewerPageAdapter
import org.worshipsongs.adapter.SongContentPortraitViewerPageAdapter
import org.worshipsongs.component.SlidingTabLayout
import org.worshipsongs.domain.Setting
import org.worshipsongs.fragment.ISongContentPortraitViewFragment
import org.worshipsongs.service.PresentationScreenService
import org.worshipsongs.service.UserPreferenceSettingService

/**
 * @Author : Seenivasan, Madasamy, Vignesh Palanisamy
 * @Version : 1.0
 */
class SongContentViewActivity : AbstractAppCompactActivity()
{
    private var userPreferenceSettingService: UserPreferenceSettingService? = null
    private var presentationScreenService: PresentationScreenService? = null
    private var isSectionView = true
    private var isTabView = true

    public override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.song_content_view)
        initSetup(savedInstanceState)
        setView()
    }

    private fun initSetup(savedInstanceState: Bundle?)
    {
        userPreferenceSettingService = UserPreferenceSettingService()
        presentationScreenService = PresentationScreenService(this)
        if (userPreferenceSettingService!!.isKeepAwake)
        {
            window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        }
        if (savedInstanceState != null)
        {
            isSectionView = savedInstanceState.getBoolean("isSectionView")
            isTabView = savedInstanceState.getBoolean("isTabView")
        }
    }

    private fun setView()
    {
        if (Configuration.ORIENTATION_PORTRAIT == resources.configuration.orientation)
        {
            setPortraitView()
        } else
        {
            setLandscapeView()
        }
    }

    private fun setPortraitView()
    {
        setActionBar()
        val songContentPortraitViewPagerAdapter = SongContentPortraitViewerPageAdapter(supportFragmentManager, intent.extras!!, presentationScreenService!!)
        // Assigning ViewPager View and setting the adapter
        val pager = findViewById<View>(R.id.pager) as ViewPager
        pager.adapter = songContentPortraitViewPagerAdapter
        // Assiging the Sliding Tab Layout View
        val tabs = findViewById<View>(R.id.tabs) as SlidingTabLayout
        //tabs.setVerticalScrollbarPosition();
        tabs.setDistributeEvenly(false)
        // To make the Tabs Fixed set this true, This makes the tabs Space Evenly in Available width
        // Setting Custom Color for the Scroll bar indicator of the Tab View
        tabs.setCustomTabColorizer(object : SlidingTabLayout.TabColorizer
        {
            override fun getIndicatorColor(position: Int): Int
            {
                return resources.getColor(android.R.color.background_dark)
            }
        })

        tabs.visibility = View.GONE
        // Setting the ViewPager For the SlidingTabsLayout
        tabs.setViewPager(pager)
        pager.currentItem = Setting.instance.position
        pager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener
        {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int)
            {
                Setting.instance.position = position
            }

            override fun onPageSelected(position: Int)
            {
                val fragment = songContentPortraitViewPagerAdapter.instantiateItem(pager, position) as ISongContentPortraitViewFragment
                fragment?.fragmentBecameVisible()
            }

            override fun onPageScrollStateChanged(state: Int)
            {
                //DO nothing when page scrolled
            }
        })
    }

    private fun setActionBar()
    {
        if (supportActionBar == null)
        {
            setCustomActionBar()
        }
    }

    private fun setLandscapeView()
    {
        val intent = intent
        val titleList = intent.extras!!.getStringArrayList(CommonConstants.TITLE_LIST_KEY)
        val songContentLandScapeViewerPageAdapter = SongContentLandScapeViewerPageAdapter(supportFragmentManager, titleList!![Setting.instance.position])
        // Assigning ViewPager View and setting the adapter
        val pager = findViewById<View>(R.id.land_pager) as ViewPager
        pager.adapter = songContentLandScapeViewerPageAdapter
        // Assiging the Sliding Tab Layout View
        val tabs = findViewById<View>(R.id.land_tabs) as SlidingTabLayout
        //tabs.setVerticalScrollbarPosition();
        tabs.setDistributeEvenly(false)
        // To make the Tabs Fixed set this true, This makes the tabs Space Evenly in Available width
        // Setting Custom Color for the Scroll bar indicator of the Tab View
        tabs.setCustomTabColorizer(object : SlidingTabLayout.TabColorizer
        {
            override fun getIndicatorColor(position: Int): Int
            {
                return resources.getColor(android.R.color.background_dark)
            }
        })
        tabs.visibility = View.GONE
        // Setting the ViewPager For the SlidingTabsLayout
        tabs.setViewPager(pager)
        Log.i(SongContentViewActivity::class.java.simpleName, "Finished")
    }

    public override fun onResume()
    {
        super.onResume()
        presentationScreenService!!.onResume()
    }


    override fun onStop()
    {
        super.onStop()
        presentationScreenService!!.onStop()
    }

    override fun onPause()
    {
        super.onPause()
        presentationScreenService!!.onPause()
    }

    override fun onSaveInstanceState(outState: Bundle)
    {
        super.onSaveInstanceState(outState)
    }

}
